### PR TITLE
Add first proposal

### PR DESCRIPTION
**Pull request by pipo02mix**
This is the first design I came up. The values (interface) for the user will be

```
bindings:
  - role: edit
    users:
      - editor1@myco.com
      - editor2@myco.com
    groups:
      - devops
    namespaces:
      - ns1
      - ns2
  - role: admin
    users:
      - admin@myco.com
    groups:
      - adminteam
```

The bindings produced:

```
helm install --dry-run fer .
---
# Source: rbac-app/templates/clusterrolebinding.yaml
apiVersion: rbac.authorization.k8s.io/v1
kind: ClusterRoleBinding
metadata:
  labels:
    helm.sh/chart: rbac-app-0.1.0
    app.kubernetes.io/name: rbac-app
    app.kubernetes.io/instance: fer
    app.kubernetes.io/version: "0.0.1"
    app.kubernetes.io/managed-by: Helm
  name: admin-1
roleRef:
  apiGroup: rbac.authorization.k8s.io
  kind: ClusterRole
  name: admin
subjects:
- kind: User
  name: admin@myco.com
  apiGroup: rbac.authorization.k8s.io
- kind: Group
  name: adminteam
  apiGroup: rbac.authorization.k8s.io
---
# Source: rbac-app/templates/clusterrolebinding.yaml
apiVersion: rbac.authorization.k8s.io/v1
kind: RoleBinding
metadata:
  labels:
    helm.sh/chart: rbac-app-0.1.0
    app.kubernetes.io/name: rbac-app
    app.kubernetes.io/instance: fer
    app.kubernetes.io/version: "0.0.1"
    app.kubernetes.io/managed-by: Helm
  name: editor-0
  namespace: ns1
roleRef:
  apiGroup: rbac.authorization.k8s.io
  kind: ClusterRole
  name: edit
subjects:
- kind: User
  name: editor1@myco.com
  apiGroup: rbac.authorization.k8s.io
- kind: User
  name: editor2@myco.com
  apiGroup: rbac.authorization.k8s.io
- kind: Group
  name: devops
  apiGroup: rbac.authorization.k8s.io
---
# Source: rbac-app/templates/clusterrolebinding.yaml
apiVersion: rbac.authorization.k8s.io/v1
kind: RoleBinding
metadata:
  labels:
    helm.sh/chart: rbac-app-0.1.0
    app.kubernetes.io/name: rbac-app
    app.kubernetes.io/instance: fer
    app.kubernetes.io/version: "0.0.1"
    app.kubernetes.io/managed-by: Helm
  name: editor-0
  namespace: ns2
roleRef:
  apiGroup: rbac.authorization.k8s.io
  kind: ClusterRole
  name: edit
subjects:
- kind: User
  name: editor1@myco.com
  apiGroup: rbac.authorization.k8s.io
- kind: User
  name: editor2@myco.com
  apiGroup: rbac.authorization.k8s.io
- kind: Group
  name: devops
  apiGroup: rbac.authorization.k8s.io
  ```

### CODE REVIEW

File: helm/rbac-app/templates/clusterrolebinding.yaml [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+{{- end }}
+{{- if gt (len $binding.groups) 0 }}
+{{- range $group := $binding.groups }}
+- kind: Group
+  name: {{ $group }}
+  apiGroup: rbac.authorization.k8s.io
+{{- end }}
+{{- end }}
+{{- end }}
+{{- end }}
```
Comment (anvddriesch) [inline helm/rbac-app/templates/clusterrolebinding.yaml:60]:
```suggestion
{{- end }}

```

**Review by anvddriesch — APPROVED** [review]
Nice!
Let's see how this works and then iterate with any additions we need.

**Comment by pipo02mix** [comment]
I have pinged @giantswarm/sig-ux for feedback

File: helm/rbac-app/values.yaml [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+name: rbac
+namespace: kube-system
+serviceType: managed
+
+project:
+  branch: "[[ .Branch ]]"
+  commit: "[[ .SHA ]]"
+
+image:
```
Comment (ubergesundheit) [inline helm/rbac-app/values.yaml:11]: I would remove these unused values and add a values.schema.json https://intranet.giantswarm.io/docs/organizational-structure/teams/cabbage/app-updates/helm-values-schema/

File: helm/rbac-app/Chart.yaml [hunk 3]
```diff
@@ -0,0 +1,10 @@
+apiVersion: v2
+appVersion: 0.0.1
+annotations:
+  application.giantswarm.io/team: "rainbow"
+  config.giantswarm.io/version: 1.x.x
+name: rbac-app
+description: A Helm chart for rbac
+engine: gotpl
+home: https://github.com/giantswarm/rbac-app
+version: 0.1.0
```
Comment (ubergesundheit) [inline helm/rbac-app/Chart.yaml:10]: You either need to use the architect templates `[[ ]]` or switch to `executor: app-build-suite` in the circle-ci config

File: helm/rbac-app/Chart.yaml [hunk 4]
```diff
@@ -0,0 +1,10 @@
+apiVersion: v2
+appVersion: 0.0.1
+annotations:
+  application.giantswarm.io/team: "rainbow"
+  config.giantswarm.io/version: 1.x.x
+name: rbac-app
+description: A Helm chart for rbac
```
Comment (ubergesundheit) [inline helm/rbac-app/Chart.yaml:7]: I know a few people in the company that won't be happy about this generic description

File: .circleci/config.yml [hunk 5]
```diff
@@ -8,10 +8,10 @@ workflows:
       - architect/push-to-app-catalog:
           context: "architect"
           # executor: "app-build-suite" # uncomment this if you want automatic metadata generation and helm chart linting
-          name: "package and push {APP-NAME} chart"
+          name: "package and push rbac chart"
           app_catalog: "giantswarm-playground-catalog"
```
Comment (ubergesundheit) [inline .circleci/config.yml:12]: is the playground catalog the right one?

File: helm/rbac-app/values.yaml (hunk 2 above)
Comment (pipo02mix) [reply helm/rbac-app/values.yaml:11]: This is coming from template app, but I guess here makes no sense

Comment (pipo02mix) [reply helm/rbac-app/values.yaml:11]: Also the binding will be commented, it is an example for now

File: .circleci/config.yml (hunk 5 above)
Comment (pipo02mix) [reply .circleci/config.yml:12]: I don't think so, but I don't know which one it will be. I will add it to official one as it is only configuration

**Comment by ubergesundheit** [comment]
Thanks for addressing my comments..

I took the liberty to enable building this app on CircleCI (on page https://app.circleci.com/projects/project-dashboard/github/giantswarm/)

Can you push an empty commit or some change to this branch to trigger a build?

**Comment by pipo02mix** [comment]
@ubergesundheit tests are green :)

**Review by ubergesundheit — APPROVED** [review]
LGTM

**Pull request merged**